app/models/data_resources/event_record.rb

```diff
@@ -141,6 +159,15 @@ def compareable_attributes
 
   # @return [Date]
   def list_date
+    if recurring? && date.present?
+      return today_in_time_range(
+        OpenStruct.new(
+          date_start: date.date_start,
+          date_end: date.date_end
+        )
+      )
+    end
+
```

Comment:
I think I can even refactor `list_date` a bit more. If there is a `date` present, we do not need to check all `dates` for the `list_date`, as we only need to consider the one date in the particular event scope!